javers-core/src/main/java/org/javers/core/graph/LiveCdoFactory.java

```diff
@@ -19,10 +20,20 @@ public LiveCdoFactory(GlobalIdFactory globalIdFactory, ObjectAccessHook objectAc
     }
 
     @Override
-    public CdoWrapper create (Object wrappedCdo, OwnerContext owner){
+    public CdoWrapper create(Object wrappedCdo, OwnerContext owner) {
+        return create(wrappedCdo, owner, false);
+    }
+
+    @Override
+    public CdoWrapper create(Object wrappedCdo, OwnerContext owner, boolean shallowReference) {
         Object wrappedCdoAccessed = objectAccessHook.access(wrappedCdo);
         GlobalId globalId = globalIdFactory.createId(wrappedCdoAccessed, owner);
-        return new CdoWrapper(wrappedCdoAccessed, globalId, typeMapper.getJaversManagedType(wrappedCdoAccessed.getClass()));
+        ManagedType managedType = typeMapper.getJaversManagedType(wrappedCdoAccessed.getClass());
+
+        if (shallowReference) {
```

Comment:
poniższy zapis wydaje mi się trochę czytelniejszy ale to tylko taka luźna uwaga

return shallowReference ?
    new ShallowCdoWrapper(wrappedCdoAccessed, globalId, managedType) : 
    new CdoWrapper(wrappedCdoAccessed, globalId, managedType);